### PR TITLE
bpo-42840: Document providing kwargs to type.

### PR DESCRIPTION
**Pull request by esoma**
https://bugs.python.org/issue42840

### CODE REVIEW

File: Doc/library/functions.rst [hunk 1]
```diff
@@ -1718,6 +1718,12 @@ are always available.  They are listed here in alphabetical order.
 
    See also :ref:`bltin-type-objects`.
 
+   Keyword arguments provided to the three argument form are passed to the
+   appropriate metaclass machinery in the same way that keywords in a class
```
Comment (Fidget-Spinner) [inline Doc/library/functions.rst]: IMO, appropriate metaclass machinery seems slightly nebulous and a user might have to search up or guess how passing keywords to classes work. Perhaps include a mention of ``object.__init_subclass__`` somewhere?
```suggestion
   appropriate metaclass machinery (usually :meth:`~object.__init_subclass__`)
   in the same way that keywords in a class
```

File: Doc/library/functions.rst [hunk 2]
```diff
@@ -1690,7 +1690,7 @@ are always available.  They are listed here in alphabetical order.
 
 
 .. class:: type(object)
-           type(name, bases, dict)
+           type(name, bases, dict, **kwds)
```
Comment (Fidget-Spinner) [inline Doc/library/functions.rst:1693]: Personally I like kwds more, but the rest of the document seems to be using ``**kwargs`` or ``**kwarg`` (there's one ``**kws`` too !), so I'd suggest using ``**kwargs`` to keep consistent.

```suggestion
           type(name, bases, dict, **kwargs)
```

**Review by Fidget-Spinner — COMMENTED** [review]
Thanks for making this PR to improve the docs! I have some minor comments.

File: Doc/library/functions.rst (hunk 2 above)
Comment (esoma) [reply Doc/library/functions.rst:1693]: That was my initial choice, but typeshed and the section on customizing class creation in the docs mostly uses `**kwds` (it also uses `**kwargs` once). Given that reasoning, if you still think `**kwargs` is best I'm happy to change it.

Comment (Fidget-Spinner) [reply Doc/library/functions.rst:1693]: Up to you. I think as it stands right now it's okay :).

**Review by Fidget-Spinner — APPROVED** [review]
LGTM!

**Review by gvanrossum — APPROVED** [review]
I'll merge this now. I'll mark it for backport to 3.9 and 3.8, but if those backports fail I'll just give up on them.

**Pull request merged**